addons/web/static/src/js/fields/special_fields.js

```diff
@@ -192,9 +196,84 @@ var FieldReportLayout = relational_fields.FieldMany2One.extend({
 });
 
 
+const IframeWrapper = AbstractField.extend({
+    description: _lt("Wrap raw html within an iframe"),
+
+    // If HTML, don't forget to adjust the sanitize options to avoid stripping most of the metadata
+    supportedFieldTypes: ['text', 'html'],
+
+    /**
+     * Note:
+     * In case of change of behaviour for this widget, please note some styles for the BaseDocumentPreview is
+     * stored here: addons/web/static/src/scss/base_document_layout.scss and contains some style linked to this
+     * widget. Proceed with caution not to break the document layout preview.
+     */
+
+    _render: function () {
+
+        this.$el.empty();
+        this.$el.attr('id', 'o_preview_iframe_wrapper')
+
+        const $spinner = $('<i class="fa fa-spinner fa-spin fa-4x" id="o_iframe_wrapper_spinner" style="color: #875a7b; z-index: 2"></i>')[0]
```

Comment:
@Arcasias Can use elaborate on "html ids is dangerous" ? I once read it could hurt performances somehow, but dangerous ? How ?

Reply:
I don't know about performances, just that when using ids you have to be sure to use a unique string. When doing so, we usually use a prefix and a unique ID set on the component/widget (see the [Owl Datepicker](https://github.com/odoo/odoo/blob/master/addons/web/static/src/js/components/datepicker.js) for example). A typical use case is when using input elements coupled with a label having a `for` attribute.

So perhaps not "dangerous", but it can quickly cause inconsistencies if you're not careful enough.